system-addon/lib/AutoMigrateFeed.jsm

```diff
@@ -0,0 +1,46 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+"use strict";
+
+const {utils: Cu} = Components;
+Cu.import("resource://gre/modules/XPCOMUtils.jsm");
+const {actionTypes: at, actionCreators: ac} = Cu.import("resource://activity-stream/common/Actions.jsm", {});
+
+Cu.import("resource://gre/modules/Log.jsm");
+let console = Log.repository.getLogger("AutoMigrate");
+console.addAppender(new Log.ConsoleAppender(new Log.BasicFormatter()));
+
+XPCOMUtils.defineLazyModuleGetter(this, "AutoMigrate",
+  "resource:///modules/AutoMigrate.jsm");
+XPCOMUtils.defineLazyModuleGetter(this, "Services",
+  "resource://gre/modules/Services.jsm");
+
+this.AutoMigrateFeed = class NewTabInit {
+  onAction(action) {
+    switch (action.type) {
+      case at.NEW_TAB_VISIBLE:
+        AutoMigrate.shouldShowMigratePrompt(action._target.browser).then(prompt => {
```

Comment:
Oh yes indeed. It's in autoland already https://bugzilla.mozilla.org/show_bug.cgi?id=1361286